edelphi/src/main/java/fi/metatavu/edelphi/pages/panel/admin/report/thesis/ThesisTimelineQueryReportPage.java

```diff
@@ -73,7 +76,7 @@ public Chart constructChart(ChartContext chartContext, QueryPage queryPage) {
     Double step = QueryPageUtils.getDoubleSetting(queryPage, "timeline.step");
     int type = QueryPageUtils.getIntegerSetting(queryPage, "timeline.type");
 
-    List<String> captions = new ArrayList<String>();
+    List<String> captions = new ArrayList<>();
     List<QueryReply> queryReplies = ReportUtils.getQueryReplies(queryPage, chartContext.getReportContext());
     for (double d = min; d <= max; d += step) {
       captions.add(step % 1 == 0 ? new Long(Math.round(d)).toString() : new Double(d).toString());
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Use "Long.toString" instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2131)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Use "Double.toString" instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2131)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this "Long" constructor [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2129)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this "Double" constructor [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2129)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Call the static method Long.toString(...) instead of instantiating a temporary object to perform this to string conversion. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1158)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Call the static method Double.toString(...) instead of instantiating a temporary object to perform this to string conversion. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1158)